core/unmarshal/unmarshal.go

```diff
@@ -0,0 +1,31 @@
+package unmarshal
+
+import (
+	"errors"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+)
+
+var errInvalidHeaderType = errors.New("invalid header type")
+
+// GetHeaderFromBytes will unmarshal the header bytes based on the header type
+func GetHeaderFromBytes(marshaller marshal.Marshalizer, headerType core.HeaderType, headerBytes []byte) (data.HeaderHandler, error) {
```

Comment:
usually, we should have a switch in a factory. Why not using the abstract factory pattern here? 
Something like this function will get an interface that can create empty objects, instead of the header type. Then, we should have 3 implementations for all types of the header.

Reply:
This is actually a kind of abstract factory pattern: https://refactoring.guru/design-patterns/abstract-factory/go/example
I don't think there is any need to over-extend this func. This is very similar to the funcs we use in `mx-chain-go` to unmarshall a shard header from bytes, e.g: `UnmarshalShardHeader`. The only difference here is that we get an extra param: `headerType` which will help us know exactly which type of header we need to unmarshall, instead of "trial and error"